python/mlc_chat/compiler/compiler_pass/pipeline.py

```diff
@@ -31,14 +34,22 @@ def transform_module(self, mod: IRModule, _ctx: tvm.transform.PassContext) -> IR
 
 
 @register_pipeline("mlc_llm")
-def _mlc_llm_pipeline():
+def _mlc_llm_pipeline(
+    variable_bounds: Dict[str, int],
+    additional_tirs: Dict[str, tvm.tir.PrimFunc],
+    metadata: Dict[str, Any],
+    skip_gemm: bool = False,
```

Comment:
@jinhongyii This is the ideal shape of a compilation pipeline, where its parameters are loosely decoupled from the rest part of the project so that it's more reusable. Might be related to your PR: https://github.com/mlc-ai/mlc-llm/pull/1380